### ISSUE TITLE
FSR for Minecraft Java

### ISSUE THREAD

**Issue by SuperFurias** [sp614x/optifine#6398]
## Description of the new feature

Regarding the AMD FSR, from what I know, is open source, so it should not be a problem adding it, and it works almost on every GPU, which is Nvidia or AMD.

**I ask forgiveness for the bad English**

**Comment by sp614x** [comment]
AMD FSR and nVidia DLSS are smart upscaling methods that mostly useful for games using complex visual effects.
They are not going to be useful in Minecraft which has almost no special effects and is mostly CPU/vertex limited.

**Comment by Mar2ck** [comment]
Surely it would be useful when running Minecraft with shader packs since then the performance will be GPU-limited in most cases. Please reconsider this feature for those of us running shaders who want better performance.

**Comment by SuperFurias** [comment]
and even nvidia DLSS is open source

**Comment by fayer3** [comment]
> and even nvidia DLSS is open source

I think you are confusing it with NVIDIA image scaling, as far as I know DLSS is still closed source.

**Comment by SuperFurias** [comment]
> > and even nvidia DLSS is open source
> 
> I think you are confusing it with NVIDIA image scaling, as far as I know DLSS is still closed source.

![image](https://user-images.githubusercontent.com/34806112/148255739-ff55a0ac-2d5e-4ef5-84cf-d21be53429b4.png)

**Comment by SuperFurias** [comment]
in combo with the shaders it would be the top

**Comment by Mar2ck** [comment]
DLSS is publicly availiable but it's closed source. NIS is the one that's open source and similar to FSR https://github.com/NVIDIAGameWorks/NVIDIAImageScaling

**Comment by pahorema** [comment]
Why was this **immediately** closed? Shaders are a supported feature and game can be very demanding with those enabled.

**Comment by SuperFurias** [comment]
> Why was this **immediately** closed? Shaders are a supported feature and game can be very demanding with those enabled.

Well, this is true, there are MANY texture pack in combo with shaders that even a 3090 has difficulties to give high fps, but hey, this is the world

**Comment by EysseW** [comment]
This would be so useful. My GPU is strugging to get good FPS with shaders...

**Comment by HyperSoop** [comment]
Please stop guys. Minecraft Java **is not a fill-rate limited game**, it's a hugely vertex-limited game, which makes implementing FSR totally not worth the time and effort. You have no idea what you are talking about.

**Comment by Mar2ck** [comment]
> which makes implementing FSR totally not worth the time and effort. You have no idea what you are talking about.

FSR1 is simply a spacial upscaler. It's designed to be a drop-in replacement for bilinear scaling, something which is already used by the vast majority of games including Minecraft and this mod. When you go into the shader settings in Optifine and turn "Rendering Quality" to below 1x, that's the game being upscaled to the screen with bilinear from a lower resolution. And when you're using GPU intensive shaders this DOES make a significant improvement to framerate. The problem is that bilinear is blurry looking. 

Enter FSR1, which functions identically to bilinear except that the upscaled picture looks much sharper and more attractive looking. The performance improvement is already there and you can try it out right now, FSR1 would just make the game look better while you were doing it.

**Comment by fayer3** [comment]
the render quality scaling is handled by the current shaderpack, not optifine itself.

**Comment by arcadeblood** [comment]
> AMD FSR and nVidia DLSS are smart upscaling methods that are mostly useful for games using complex visual effects. They are not going to be useful in Minecraft which has almost no special effects and is usually CPU/vertex limited.

1st, It's very easy for dlss algoritm to count cubes.
2nd If minecraft is soo not demanding, try to keep your framerate high with seus ptgi hrr 2.1 and 512x512 texturepack.
You'd need at least rtx 3060 ti. With raytracing set up for the lowest of the lows.
3rd You literally have dlss in "windows edition". And it works pretty well, with anything you will throw at it.
4th I understand difficulty in implementing dlss, but fsr should have no problem being there.

**Comment by SuperFurias** [comment]
even a 4090 has a hard time keeping the game in 4K over 60 fps with SEUS PTGI HRR 2.1, 64 Chunks /w RP
there was a time where the game hitted 50 fps. i really think that DLSS needs to be implemented inside optifine, not only for people who want to use extreme shaders and settings, but even for normal gaming, having more fps is always better

**Comment by SuperFurias** [comment]
and NVIDIA DLSS is completely open source 
https://github.com/NVIDIA/DLSS

**Comment by IMS212** [comment]
DLSS is not open source. The SDK is, which allows any supported library (not OpenGL!) to use DLSS. The actual DLSS code is in a single DLL file which is unchangeable.

**Comment by IMS212** [comment]
tl;dr in case people aren’t getting it: sp is legally not allowed to backport DLSS to OpenGL, and a Vulkan port is currently not in the cards

**Comment by arcadeblood** [comment]
> tl;dr in case people aren’t getting it: sp is legally not allowed to backport DLSS to OpenGL, and a Vulkan port is currently not in the cards

 you can already use fsr for opengl. You have magpie and lossless scaling [programs]
and if nvidia wants to fight with modding community, then good freaking luck.

**Comment by LiberteIV** [comment]
Hmm that's sad that this request was closed.
I agree with Seus PTGI HRR Shader and Patrix Resourcepack the frames are not good.
AMD FidelityFX Super Resolution could surely take that heavly load of the GPU, especially the new 2.1.2 version is amazing, in pleague tale requiem my fps went from 40 with native to 80 with FSR 2.

I'f also checked a fabri mod which replaces OpenGL with Vulkan, as a result the average FPS was increased by massive 400-600.
If optifine was to support vulkanAPI and FSR 2.1.2 imagine the performance increase . . . what a dream.

**Comment by LiberteIV** [comment]
> Surely it would be useful when running Minecraft with shader packs since then the performance will be GPU-limited in most cases. Please reconsider this feature for those of us running shaders who want better performance.

Exactly my toughts...

**Comment by EysseW** [comment]
How about FSR though?

**Comment by Artoria2e5** [comment]
Three annoying cents on this thing.
* Putting it in Optifine (or any Minecraft mod) has the main advantage of offering depth info. If we don’t go for it why don’t we use Magpie?
* FSR2 is indeed a good idea, if someone ports it to GL (which is legal)  https://github.com/GPUOpen-Effects/FidelityFX-FSR2/issues/10. But nobody has done that yet. Maybe it can be in a shaderpack, who knows.
* DLSS is out of the question per above, as we would need to do DLL distribution.

**Comment by Quest79** [comment]
Did [sp614x](https://github.com/sp614x) have a brainfart and then run away? What the hell was up with his reply? Did he forget the mod he himself made supports shaders? Of course we wouldent be considering DLSS without shaders??? Duh?

**Comment by Creeper199** [comment]
fsr2 is compatible white vulkan (for what i now), the is a mod that rewrites Minecraft OpenGL renderer to use Vulkan API.
(the mod name is ''Vulkan render'' it is for fabric)

**Comment by nellydocs** [comment]
https://github.com/JuanDiegoMontoya/FidelityFX-FSR2-OpenGL

leaving this here, it might help someone somewhere sometime :)

**Comment by netflexy** [comment]
> Please stop guys. Minecraft Java **is not a fill-rate limited game**, it's a hugely vertex-limited game, which makes implementing FSR totally not worth the time and effort. You have no idea what you are talking about.

you have no idea, what THEY talking about

**Comment by Legendile7** [comment]
Still nothing in 2024 rip.

**Comment by HColeman49** [comment]
It's a joke that they closed it before getting any debate/reasoning. For starters, seus PEGI HRR series (1, 2, 2.1 e.t.c) of shaders include upscaling like FSR and the original version is seus E12 or E13, I can't remember but its one of those two. If you don't believe me, look at his blog on patron. It still would be a nice idea to have fsr and dlss on java for packs that do not have this feature, however sadly this will most likely never happen (well... Untill FSR 3.0 comes out, supporting all games late this year or next). Myself as a game designer, I don't believe changing the rendering engine from what it is supposed to be is a smart idea as there will be a lot of features that might rely on the current engine. Also, using mods to do this might effect other mods like discussed earlier in a previous comment as there would be two rendering engines present. Even IF it worked, the amount of graphical errors and issues would be ridiculous... It took Seus multiple years of development to get this shaders to the point they are now, and they still have issues...

Summary :
Wait till FSR 3.0 comes out (hopefully it supports minecrafts rendering engine)

Don't have your hopes up for a built in upscaler (unless forge ever supports changing the rendering engine)

**Comment by IMS212** [comment]
It's not closed because no one cares; it's because it's too hard. If it's easy, do it yourself.

**Comment by IMS212** [comment]
No, that is not how it works. (For reference,I make Iris.) A new engine would basically be saying to rewrite Minecraft; and at that point, why are you playing the game?

**Comment by HColeman49** [comment]
> No, that is not how it works. (For reference,I make Iris.) A new engine would basically be saying to rewrite Minecraft; and at that point, why are you playing the game?

Well done on your achievement with iris. However a launcher is much different than a mod. You never know. I'm not knowledgeable about it but I believe everything is somewhat possible. Let's just come to a agreement that it is not possible. Just wait till FSR 3.0 comes out, or rely on shaders to provide their own upscaling methods like seus 's hrr tests

**Comment by IMS212** [comment]
Um... OptiFine and Iris are both mods; they both modify the same part of Minecraft. Same as Forge.

**Comment by HColeman49** [comment]
I agree with you... No point pushing it further than it is.

**Comment by HColeman49** [comment]
It does confuse me why microsoft doesn't provide multiple rendering engines for performance. A fair amount of games change their rendering engine based on the compatability of the hardware playing the game, however microsoft doesn't want to do this with minecraft. This would still allow people on MacOS and other operating systems to still play the game, whilst giving other people more fps. To combat what I just wrote, apparently minecraft uses a form of OpenGl, called Blaze3D, which could be a library that is solely dependent on OpenGl. If this is the case, they would have to change the engine AGAIN like they have already in 1.15. I understand why they closed this form, this is never going to happen.

**Comment by Legendile7** [comment]
I mean this is Mojang you're talking about, the team that gives us "big" updates every year that would take a couple of developers a week to make.

**Comment by cln100** [comment]
The "Magpie" software seems to have solved the problem of applying FSR scaling, as it works with all physical programs. Of course, there are other forms of scaling in this software, which are also different in compatibility and picture quality. The problem with DLSS seems to be something that no one can solve, and that's obvious.
(Please forgive me for not being able to continue leaving messages, after all, China needs to turn on VPN to access github)
![Uploading 屏幕截图 2024-07-24 110653.png…]()

**Comment by reallynniroprobably** [comment]
> > Why was this **immediately** closed? Shaders are a supported feature and game can be very demanding with those enabled.
> 
> Well, this is true, there are MANY UHD texture pack in combo with shaders where even a 3090 has difficulties to give high fps, but hey, this is the world
i totally agree and i hate it when people are like "you don't need this feature, its minecraft".
they struggle to realise that minecraft is spagetti coded and needs a suprising amount of graphics power for good framerates on even dedicated gaming computers
